syntax = "proto2";

package aws;


// NextId: 8
message Model {
  // NextId: 7
  enum ModelOption {
    UNKNOWN = 0;

    // Claude 2.
    AWS_CLAUDE_2 = 1;
    AWS_CLAUDE_2D1 = 2;

    // Claude 3.
    AWS_CLAUDE_3 = 3;
    AWS_CLAUDE_3_HAIKU = 4;
    AWS_CLAUDE_3_OPUS = 5;
    AWS_CLAUDE_3_SONNET = 6;
    AWS_CLAUDE_35_SONNET = 7;
    AWS_CLAUDE_35_V2_SONNET = 8;
    AWS_NOVA_PRO = 9;
    AWS_LLAMA_31_70B = 10;
    AWS_MISTRAL_LARGE_2 = 11;
    AWS_CLAUDE_35_HAIKU = 12;
    US_AWS_LLAMA_31_70B = 13;
    US_AWS_CLAUDE_35_V2_SONNET = 14;
  }

  // Model name or option can be used to infer model_id.
  oneof model {
    string model_id = 1;
    ModelOption model_option = 2;
  }

  optional int32 top_k = 3;        // [0, 100M]
  optional float top_p = 4;        // [0, 1]: Default is 0.999
  optional float temperature = 5;  // [0, 1]: Default is 1

  optional int32 max_tokens = 6 [default = 1024];
  optional string anthropic_version = 7 [default = "bedrock-2023-05-31"];
}
